File: student_manager.py
print("=== Gestor de Datos de Estudiante - Python ===")

# Solicitar datos
nombre = input("Ingrese el nombre del estudiante: ")
edad = int(input("Ingrese la edad: "))
promedio = float(input("Ingrese el promedio: "))
activo = input("¿Está activo? (sí/no): ").lower() == 'sí'

# Mostrar datos
print("\nDatos del estudiante:")
print(f"Nombre: {nombre}")
print(f"Edad: {edad}")
print(f"Promedio: {promedio}")
print(f"Activo: {'Sí' if activo else 'No'}")

# Operaciones
edad += 1
promedio_entero = round(promedio)
estado = "Activo" if activo else "Inactivo"

print("\nDespués de operaciones:")
print(f"Edad incrementada: {edad}")
print(f"Promedio redondeado: {promedio_entero}")
print(f"Estado: {estado}")

# Mostrar tipos
print("\nTipos de datos:")
print(f"Nombre: {type(nombre)}")
print(f"Edad: {type(edad)}")
print(f"Promedio: {type(promedio)}")
print(f"Activo: {type(activo)}")

# Casos especiales
print("\nCasos especiales:")
try:
    edad_decimal = 20.5
    edad_int = int(edad_decimal)
    print(f"Asignar decimal {edad_decimal} a entero: {edad_int}")
except Exception as e:
    print(f"Error: {e}")

try:
    suma = 5 + "10"
except Exception as e:
    print(f"Sumar número + string: Error - {e}")

# En Python, variable sin inicializar da error
print("Usar variable sin inicializar: Error (NameError)")

print("\nPrograma completado.")

File: student_manager.c
#include <stdio.h>
#include <string.h>

struct Estudiante {
    char nombre[100];
    int edad;
    float promedio;
    int activo; // 1 true, 0 false
};

int main() {
    struct Estudiante est;
    
    printf("=== Gestor de Datos de Estudiante - C ===\n");
    
    printf("Ingrese el nombre: ");
    fgets(est.nombre, 100, stdin);
    est.nombre[strcspn(est.nombre, "\n")] = 0; // remove newline
    
    printf("Ingrese la edad: ");
    scanf("%d", &est.edad);
    
    printf("Ingrese el promedio: ");
    scanf("%f", &est.promedio);
    
    char activo_str[10];
    printf("¿Está activo? (si/no): ");
    scanf("%s", activo_str);
    est.activo = (strcmp(activo_str, "si") == 0 || strcmp(activo_str, "sí") == 0);
    
    // Mostrar datos
    printf("\nDatos del estudiante:\n");
    printf("Nombre: %s\n", est.nombre);
    printf("Edad: %d\n", est.edad);
    printf("Promedio: %.2f\n", est.promedio);
    printf("Activo: %s\n", est.activo ? "Sí" : "No");
    
    // Operaciones
    est.edad++;
    int prom_entero = (int)(est.promedio + 0.5);
    char* estado = est.activo ? "Activo" : "Inactivo";
    
    printf("\nDespués de operaciones:\n");
    printf("Edad incrementada: %d\n", est.edad);
    printf("Promedio redondeado: %d\n", prom_entero);
    printf("Estado: %s\n", estado);
    
    // Tipos (sizeof)
    printf("\nTamaños (sizeof):\n");
    printf("Nombre: %zu bytes\n", sizeof(est.nombre));
    printf("Edad: %zu bytes\n", sizeof(est.edad));
    printf("Promedio: %zu bytes\n", sizeof(est.promedio));
    printf("Activo: %zu bytes\n", sizeof(est.activo));
    
    // Casos
    printf("\nCasos especiales:\n");
    float dec = 20.5;
    int entero = (int)dec;
    printf("Asignar decimal a entero: %d (con casting)\n", entero);
    
    // Sumar numero + string not possible easily
    printf("Sumar número + string: No directo, requiere manejo manual.\n");
    
    // Variable sin inicializar
    int sin_init;
    printf("Variable sin inicializar: valor garbage %d (undefined behavior)\n", sin_init);
    
    return 0;
}